src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -1881,8 +1881,9 @@ subroutine ensemble_forward_model(cvec,a_en,ibin)
              do ig=1,ntotensgrp
                 iaens=ensgrp2aensgrp(ig,ic2+nc3d,ibin)
                 if(iaens>0) then
-                   do n=1,n_ens
-                      do j=1,jm
+!$omp parallel do schedule(static,1) private(j,n,k,i)
```

Comment:
What is the benefit of flipping the order of n and j here? Should the n=1,n_ens be on the outer loop for it is on the dimension that changes slower in their storage?

Reply:
It is so that it can be threaded over j.

Reply:
Thanks.  In current gsi runs,  the used openmp threads are normally than the standard ensemble size (n_ens=30 for rrfs and 80 for gfs), so, i think using multiple threads on n is fine while avoiding potential issue i mentioned above. 
And, shall we add openmp to other places (not only here for case(ps).)?    but I understand this is a work supposedly planed to be done in the future.   So, just please make sure the flipping does show benefits (theoretically or practically).  Thanks.

Reply:
This part of the code cannot be threaded over the number of ensembles (without a lot of extra work) since it is summing into an array that is not dependent on n.

However, this code change probably does not do much in terms of optimization.  So I have removed the threading and put the loop order at n,k,j,i.

Reply:
Thanks. I missed the operation dependency for loop over n_ens.  And, also agree maybe reverting to the original order is better before extra efforts on this , which will be very helpful for optimization of all those "huge" loops for various variables (like treatment for the above  n_ens dependency).